File: blog_backend/littleseventhirty-blog/src/main/java/cn/edu/tjufe/zql/domain/entity/Category.java
package cn.edu.tjufe.zql.domain.entity;

import cn.edu.tjufe.zql.domain.ViewObjectConvertible;
import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

/**
 * @author: littleseventhirty
 * @description: 分类实体
 * @date: 2025/12/15-16:15
 **/
@Data
@AllArgsConstructor
@NoArgsConstructor
@TableName("t_category")
public class Category implements ViewObjectConvertible {
    // 数据库标识
    private Long id;
    // 分类ID
    private Long categoryId;
    // 分类名称
    private String categoryName;
    // 创建时间
    @TableField(fill = FieldFill.INSERT)
    private Date createTime;
    // 更新时间
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Date updateTime;
    // 是否删除（0：未删除，1：已删除）
    private Integer isDeleted;
}

File: blog_frontend/littleseventhirty-blog/src/mock/category.ts
import Mock from 'mockjs'

// 1. 定义接口 URL
// 使用正则匹配，以防止 url 中后续带参数导致匹配失败
// 记得加上拦截器添加的 /api 前缀
const categoryListUrl = /\/api\/category\/list/

// 2. 定义 Mock 规则
Mock.mock(categoryListUrl, 'get', {
  "code": 200, // 对应统一返回格式的 code
  "msg": "操作成功", // 对应 msg
  // data 对应 List<CategoryVO>，生成 5 到 15 条随机数据
  "data|5-15": [
    {
      // categoryId: Long -> 使用自增数字或随机 ID
      "categoryId|+1": 100,

      // categoryName: String -> 使用随机中文标题 (3-6字)
      "categoryName": "@ctitle(3, 6)",

      // articleCount: Long -> 随机整数
      "articleCount": "@integer(0, 500)",

      // createTime: Date -> 格式化日期字符串
      "createTime": "@datetime('yyyy-MM-dd HH:mm:ss')",

      // updateTime: Date -> 格式化日期字符串
      "updateTime": "@datetime('yyyy-MM-dd HH:mm:ss')"
    }
  ]
})